src/Spec.ts

```diff
@@ -503,6 +507,129 @@ export default class Spec {
     }
   }
 
+  private async buildMultipage() {
+    this.log('Creating multipage build...');
+
+    const doc = this.doc.cloneNode(true) as Document;
+
+    const links = new Map<string, string>();
+    const linksForJson: { [key: string]: string[] } = {};
+    const registerLinks = (page: string, node: Element) => {
+      const all = [];
+      if (node.id) {
+        links.set(`#${node.id}`, `${page}#${node.id}`);
+        all.push(node.id);
+      }
+      Array.from(node.querySelectorAll('[id]')).forEach(n => {
+        links.set(`#${n.id}`, `${page}#${n.id}`);
+        if (!n.id.startsWith('_ref')) {
+          all.push(n.id);
+        }
+      });
+      linksForJson[page] = all;
+    };
+
+    const pages = new Map<string, Element>();
+    pages.set('index', doc.createElement('div'));
+    let hasIntro = false;
+    Array.from(doc.querySelector('#spec-container')!.children).forEach(c => {
+      if (!hasIntro) {
+        registerLinks('index.html', c);
+        pages.get('index')!.appendChild(c);
+        if (c.nodeName === 'EMU-INTRO') {
+          hasIntro = true;
+        }
+      } else {
+        if (!c.id) {
+          throw new Error('Element cannot be used as page because it is missing an id');
+        }
+        pages.set(c.id, c);
+        registerLinks(`${c.id}.html`, c);
+      }
+    });
+    if (!hasIntro) {
+      this.warn({
+        type: 'global',
+        ruleId: 'multipage-intro',
+        message: 'Multipage build was enabled but no introduction was found.',
+      });
+    }
+
+    // hack: add index div to document so below rewrites can reach it.
+    // don't perform restructuring after this point.
+    doc.body.appendChild(pages.get('index')!);
+
+    Array.from(doc.querySelectorAll('a')).forEach(a => {
+      const href = a.getAttribute('href')!;
+      if (links.has(href)) {
+        a.setAttribute('href', links.get(href)!);
+      }
+    });
+
+    [
+      ['img', 'src'],
+      ['object', 'data'],
+      ['script', 'src'],
+      ['link', 'href'],
+    ].forEach(([type, attr]) => {
+      Array.from(doc.querySelectorAll(type)).forEach(e => {
+        const val = e.getAttribute(attr);
+        if (val && !val.startsWith('http')) {
+          e.setAttribute(attr, `../${val}`);
+        }
+      });
+    });
+
+    const REWRITE_SCRIPT = `\
+'use strict';
+
+if (document.location.hash) {
+  const obj = JSON.parse(${JSON.stringify(JSON.stringify(linksForJson))});
+  for (const [page, ids] of Object.entries(obj)) {
+    if (ids.includes(document.location.hash.slice(1))) {
+      window.location.href = page + document.location.hash;
+      break;
+    }
+  }
+}
+`;
+
+    const format = (e: Element) => {
+      return `\
+<!DOCTYPE html>
```

Comment:
```suggestion
<!doctype html>
```

(to match [single-page builds](https://github.com/tc39/ecmarkup/blob/40d56a311165bb974e75ab50b8803115ee560700/src/Spec.ts#L405))